src/custom/pages/Swap/components/CompatibilityIssuesWarning.tsx

```diff
@@ -0,0 +1,38 @@
+import UnsupportedCurrencyFooter from 'components/swap/UnsupportedCurrencyFooter'
+import { HashLink } from 'react-router-hash-link'
+import { Currency } from '@uniswap/sdk-core'
+
+export interface CompatibilityIssuesWarningProps {
+  currencyIn: Currency
+  currencyOut: Currency
+  isSupportedWallet: boolean
+  swapIsUnsupported: boolean
+}
+
+export function CompatibilityIssuesWarning(props: CompatibilityIssuesWarningProps) {
+  const { currencyIn, currencyOut, isSupportedWallet, swapIsUnsupported } = props
+  const currenciesPair = [currencyIn, currencyOut]
+
+  if (!swapIsUnsupported) {
+    return null
+  }
```

Comment:
The display condition for this component is split with part here and part on `currencyIn && currencyOut` condition.

Unless this will be re-used later on, I rather have those conditions in the same place.
Either add `!swapIsUnsupported` to `currencyIn && currencyOut` or add a parameter like:

shouldRender={currencyIn && currencyOut && !swapIsUnsupported}

What do you think?

Reply:
Fixed: https://github.com/cowprotocol/cowswap/pull/1049/commits/2c57ee9c6a374122ce2d0de371fc80c3dfcd76b4

I prefer to avoid the usage of optional parameters, so I left all conditions outside of the component and CompatibilityIssuesWarningProps doesn't contain optional parameters.